python/setup.py

```diff
@@ -142,23 +140,39 @@ def get_json_package_info():
 
 # llvm
 def get_llvm_package_info():
-    # added statement for Apple Silicon
     system = platform.system()
-    arch = platform.machine()
-    if arch == 'aarch64':
-        arch = 'arm64'
+    arch = {"x86_64": "x64", "arm64": "arm64", "aarch64": "arm64"}[platform.machine()]
     if system == "Darwin":
-        arch = platform.machine()
-        if arch == "x86_64":
-            arch = "x64"
         system_suffix = f"macos-{arch}"
     elif system == "Linux":
         if arch == 'arm64':
             system_suffix = 'ubuntu-arm64'
         else:
-            vglibc = tuple(map(int, platform.libc_ver()[1].split('.')))
-            vglibc = vglibc[0] * 100 + vglibc[1]
-            system_suffix = 'ubuntu-x64' if vglibc > 217 else 'centos-x64'
+            # Ubuntu 22.04 has GLIBCXX support up to GLIBCXX_3.4.30.
+            # Ubuntu 20.04 has GLIBCXX support up to GLIBCXX_3.4.28.
+            # Almalinux 8 (and other RHEL8 derivatives) have GLIBCXX
+            # support up to GLIBCXX_3.4.25.
+            target_libcxx_found = False
+            if os.path.isfile("/usr/lib/" + platform.machine() + "-linux-gnu/libstdc++.so.6"):
+                check_libcxx_version = \
+                    subprocess.run(
+                        "strings /usr/lib/"+platform.machine()+"-linux-gnu/libstdc++.so.6 | grep GLIBCXX_3.4.26", \
```

Comment:
Changing this check from GLIBCXX_3.4.30 to GLIBCXX_3.4.26 is the fix for Ubuntu 20 build failure. 

Supported GLIBCXX versions on each OS verified by spinning up Azure VMs of the respective OS'es and running the strings command on them.